### ISSUE TITLE
Generate dSYM files for LLDB

### ISSUE THREAD

**Issue by LubosD** [darlinghq/darling#403]
Steps:
* For every `add_darling_executable()` and `add_darling_library()` call, generate a (flat?) dSYM file using `llvm-dsymutil`.
* Install dSYM files somewhere, e.g. `/System/Library/Caches/dsym` preferrably using the [UUID scheme](https://lldb.llvm.org/symbols.html).
* Implement `DBGCopyFullDSYMURLForUUID()` in `DebugSymbols.framework`.
* Nice to have: support paths set via `defaults write com.apple.DebugSymbols DBGFileMappedPaths` (depends on #402).
